src/InertialSense.cpp

```diff
@@ -1025,7 +1025,7 @@ bool InertialSense::isFirmwareUpdateFinished() {
     for (int i = 0; i < (int) m_comManagerState.devices.size(); i++) {
         ISFirmwareUpdater *fwUpdater = m_comManagerState.devices[i].fwUpdater;
         if (fwUpdater != nullptr) {
-            if (fwUpdater->hasPendingCommands() || (fwUpdater->fwUpdate_getSessionStatus() < fwUpdate::NOT_STARTED) || (fwUpdater->fwUpdate_getSessionStatus() >= fwUpdate::FINISHED))
+            if (!fwUpdater->fwUpdate_isDone())
```

Comment:
This two ifs should be combined: `if (fwUpdater != null && fwUpdate->fwUpdate_isDone()) return false;`